keps/sig-auth/4317-pod-certificates/README.md

```diff
@@ -0,0 +1,1045 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary and Motivation
+
+The certificates.k8s.io API group provides a flexible, pluggable mechanism to request X.509 certificates within a Kubernetes cluster, but actually making certificates available to your workloads is left as an exercise for the reader.
+
+This KEP defines three pieces of fundamental machinery:
+* An X.509 extension for expressing pod identity
+* Node restriction support for certificate requests that use the PodIdentity
+  extension.
+* A PodCertificate projected volume source that instructs the kubelet to handle
+  provisioning a key and certificate chain on behalf of a pod.  
+Taken together, this machinery makes it feasible to securely and automatically
+deliver X.509 certificates to every pod in a cluster, without imposing an
+unreasonable burden on application developers or cluster administrators.  
+
+As a first application of this machinery, this KEP defines additional machinery that allows pods to automatically receive mTLS client certificates suitable for authenticating to kube-apiserver, iwth the same attributes provided by bound service account tokens:
+* A new TLS user-mapping capability in kube-apiserver to understand client
+  certificates that embed PodIdentity extensions.
+* A new approver/signer pair, shipped in kube-controller-manager, to issue
+  compatible client certificate to pods.
+* An enhancement to client-go's InClusterConfig that automatically loads a
+  credential bundle from a well-known filesystem path and uses is as a client
+  certificate.
+
+There is a draft implementation of the design outlined in this KEP:
+https://github.com/kubernetes/kubernetes/pull/121596
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
```

Comment:
Additional goal: revocation for the kube-apiserver pod client certificate case --- revocation handled via pod binding.

Reply:
Added a section for this.